packages/drivers/routerlicious-driver/src/requestUtils.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import Axios, { AxiosRequestConfig, AxiosResponse, AxiosError } from "axios";
+import { isStatusRetriable, throwR11sNetworkError } from "./r11sError";
+
+export async function enhanceRequestError(error: any): Promise<never> {
+    if (error?.isAxiosError) {
+        const axiosError = error as AxiosError;
+        throwR11sNetworkError(
+            axiosError.response?.statusText
+                ? `${axiosError.response?.data} (${axiosError.response?.statusText})`
+                : axiosError.response?.data,
+            axiosError.response?.status,
+            isStatusRetriable(axiosError.response?.status, axiosError.response?.data?.retryAfter),
+            axiosError.response?.data?.retryAfter
+                ? axiosError.response?.data?.retryAfter
+                : undefined,
+        );
+    }
+    throwR11sNetworkError(`${error}`);
```

Comment:
Is it guaranteed that error will always be axios error? If its not, is it always a string?

Reply:
I was unsure about this. In all the cases I'm trying to catch, yes, it should be an Axios error, but by the nature of it being an error it felt like I shouldn't assume it always would be. It would probably be safest to wrap this in json-stringify-safe